src/common/transformations/src/transformations/op_conversions/convert_avgpool_downgrade.cpp

```diff
@@ -0,0 +1,83 @@
+        ov::Shape pads_begin;
+        ov::Shape pads_end;
+
+        using ov::op::v0::Constant;
+        using ov::op::v0::Concat;
+        using ov::op::v1::Pad;
+        using ov::op::v1::Subtract;
+        using ov::op::v1::ConvertLike;
+        using ov::op::v3::Broadcast;
+        using ov::op::v3::ShapeOf;
+
+        if (!exclude_pad) {
+            const auto zero = Constant::create(element::f32, Shape{}, {0});
+            const auto zero_node = node_registry.make<ConvertLike>(zero, input);
+            const auto zero_i64 = Constant::create(element::i64, Shape{}, {0});
+            const auto shape = node_registry.make<ShapeOf>(input, element::i64);
+            const auto rank = node_registry.make<ShapeOf>(shape, element::i64);
+            const auto pads_begin_v14 = avg_pool_v14->get_pads_begin();
+            const auto pads_begin_node = Constant::create(element::i64, Shape{pads_begin_v14.size()}, pads_begin_v14);
+            const auto pads_end_v14 = avg_pool_v14->get_pads_end();
+            const auto pads_end_node = Constant::create(element::i64, Shape{pads_end_v14.size()}, pads_end_v14);
+            const auto pads_len = Constant::create(element::i64, Shape{}, {pads_begin_v14.size()});
+            const auto pads_diff = node_registry.make<Subtract>(rank, pads_len);
+            const auto pads_remaining = node_registry.make<Broadcast>(zero_i64, pads_diff);
+            const auto pads_begin_v1 =
+                node_registry.make<Concat>(OutputVector{std::move(pads_remaining), std::move(pads_begin_node)}, 0);
+            const auto pads_end_v1 =
+                node_registry.make<Concat>(OutputVector{std::move(pads_remaining), std::move(pads_begin_node)}, 0);
+            const auto pad_node =
+                node_registry.make<Pad>(input, pads_begin_v1, pads_end_v1, zero_node, ov::op::PadMode::CONSTANT);
+            pads_begin = Shape(pads_begin_v14.size(), 0);
+            pads_end = Shape(pads_begin_v14.size(), 0);
+        } else {
+            pads_begin = avg_pool_v14->get_pads_begin();
+            pads_end = avg_pool_v14->get_pads_end();
+        }
+        const auto avg_pool_v1 = node_registry.make<ov::op::v1::AvgPool>(input,
+                                                                         avg_pool_v14->get_strides(),
+                                                                         pads_begin,
+                                                                         pads_end,
+                                                                         avg_pool_v14->get_kernel(),
+                                                                         exclude_pad,
+                                                                         rounding_type_v1,
+                                                                         avg_pool_v14->get_auto_pad());
+        avg_pool_v1->set_friendly_name(avg_pool_v14->get_friendly_name());
+        copy_runtime_info(avg_pool_v14, node_registry.get());
```

Comment:
`pad_node` is not provided as `input` to the `AvgPool`, looks like not connected to the subgraph.
Is there a unit test for this `(!exclude_pad)` case?

Reply:
Fixed, the node was missing from the test.